frontend/src/metabase/querying/components/FilterValuePicker/SearchValuePicker/SearchValuePicker.tsx

```diff
@@ -35,19 +38,37 @@ export function SearchValuePicker({
   const [searchValue, setSearchValue] = useState("");
   const [searchQuery, setSearchQuery] = useState(searchValue);
 
-  const { data: fieldValues = initialFieldValues } = useSearchFieldValuesQuery(
+  const { data: searchFieldValues = initialFieldValues } =
+    useSearchFieldValuesQuery(
+      {
+        fieldId,
+        searchFieldId,
+        value: searchQuery,
+        limit: SEARCH_LIMIT,
+      },
+      {
+        skip: !searchQuery,
+      },
+    );
+
+  const { data: remappedFieldValues = [] } = useGetRemappedFieldValuesQuery(
     {
       fieldId,
-      searchFieldId,
-      value: searchQuery,
-      limit: SEARCH_LIMIT,
+      remappedFieldId: searchFieldId,
+      values: selectedValues,
     },
     {
-      skip: !searchQuery,
+      skip:
+        fieldId === searchFieldId ||
+        selectedValues.length === 0 ||
+        searchValue.length !== 0,
```

Comment:
Is this to avoid CLS while user is typing? Or to optimize?
Definitely deserves a comment.

But maybe we don't need it?